nemo/collections/asr/modules/audio_preprocessing.py

```diff
@@ -12,17 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import math
 from abc import ABC, abstractmethod
 
 import torch
+from packaging import version
 
+from nemo import logging
 from nemo.collections.asr.parts.features import FilterbankFeatures
 from nemo.collections.asr.parts.spectr_augment import SpecAugment, SpecCutout
 from nemo.core.classes import NeuralModule, typecheck
-from nemo.core.neural_types import AudioSignal, LengthsType, MelSpectrogramType, NeuralType, SpectrogramType
-from nemo.utils.decorators import experimental
-
-__all__ = ['AudioToMelSpectrogramPreprocessor', 'SpectrogramAugmentation']
+from nemo.core.neural_types import (
+    AudioSignal,
+    LengthsType,
+    MelSpectrogramType,
+    MFCCSpectrogramType,
+    NeuralType,
+    SpectrogramType,
+)
+
+try:
+    import torchaudio
+    import torchaudio.transforms
+    import torchaudio.functional
+
+    TORCHAUDIO_VERSION = version.parse(torchaudio.__version__)
+    TORCHAUDIO_VERSION_MIN = version.parse('0.5')
+
+    HAVE_TORCHAUDIO = True
+except ModuleNotFoundError:
+    HAVE_TORCHAUDIO = False
+    logging.warning('Could not import torchaudio. Some features might not work.')
```

Comment:
this is very annoying warning which is happening on every import. Can we move it under class which uses it and just raise exception is the constructor of that class (and say that torchaudio needs to be installed)

Reply:
I agree. I'll defer the logging to the moment of actually requiring it.

Reply:
Done